packages/prettier-plugin-java/src/printers/expressions.ts

```diff
@@ -324,10 +321,50 @@ export class ExpressionsPrettierVisitor extends BaseCstPrettierPrinter {
     params?: { addParenthesisToWrapStatement?: boolean }
   ) {
     const countMethodInvocation = isUniqueMethodInvocation(ctx.primarySuffix);
+    const newExpression =
+      ctx.primaryPrefix[0].children.newExpression?.[0].children;
+    const arrayCreationExpression =
+      newExpression?.arrayCreationExpression?.[0].children;
+    const classInstanceCreationExpression =
+      newExpression?.unqualifiedClassInstanceCreationExpression?.[0].children;
+    const isBreakableNewExpression =
+      countMethodInvocation <= 1 &&
+      [
+        arrayCreationExpression?.classOrInterfaceType?.[0].children.classType[0]
+          .children.typeArguments,
+        arrayCreationExpression?.arrayCreationExplicitInitSuffix?.[0].children
+          .arrayInitializer[0].children.variableInitializerList,
+        classInstanceCreationExpression?.classOrInterfaceTypeToInstantiate[0]
+          .children.typeArgumentsOrDiamond?.[0].children.typeArguments,
+        classInstanceCreationExpression?.argumentList
+      ].some(breakablePart => breakablePart);
+    const fqnOrRefType =
+      ctx.primaryPrefix[0].children.fqnOrRefType?.[0].children;
+    const fqnOrRefTypeParts = [
+      fqnOrRefType?.fqnOrRefTypePartFirst[0],
+      ...(fqnOrRefType?.fqnOrRefTypePartRest ?? [])
+    ];
+    const nextToLastIdentifier =
+      fqnOrRefTypeParts[fqnOrRefTypeParts.length - 2]?.children
+        .fqnOrRefTypePartCommon[0].children.Identifier?.[0].image;
```

Comment:
WDYT of extracting a nextToLastIdentifier function ?

Reply:
Since I extracted `isCapitalizedIdentifier` out into its own function and `nextToLastIdentifier` was only used there, I simply moved it into that function. Let me know if you'd still like that broken out further.